src/DataStructure/Trie/TrieNode.ts

```diff
@@ -0,0 +1,123 @@
+import isUndefined from "Type/isUndefined/isUndefined"
+import add from "Math/add/add"
+import map from "List/map/map"
+import reduce from "List/reduce/reduce"
+
+export default class TrieNode<T> {
+
+    keypart: string
+    value: T
+    parent: TrieNode<T>
+    children: TrieNode<T>[] = []
+
+    constructor(value: T, keypart: string, parent: TrieNode<T> = undefined) {
```

Comment:
Would be better if you would call it explicitly as an optional parameter instead.

constructor(value: T, keypart: string, parent?: TrieNode<T>)